### PR TITLE
Added some features, tests

### PR DESCRIPTION
**Pull request by delijati**
Features:

- Python3 support.
- Added tests, travisci.
- Proper flask extension.
- Global registry, but blueprint url rules.
- Add basepath if not provided.

Todo:

- registerModel should also be usable as a function add_registerModel

To release a egg i would suggest to use https://pypi.python.org/pypi/zest.releaser or https://pypi.python.org/pypi/jarn.mkrelease

### CODE REVIEW

**Comment by delijati** [comment]
To prevent from creating a ``MANIFEST.in`` try https://pypi.python.org/pypi/setuptools-git

File: flask_sillywalk/sillywalk.py [hunk 1]
```diff
@@ -111,28 +108,121 @@ def inner_func(c, *args, **kwargs):
...
+                     f,
+                     method="GET",
+                     content_type="application/json",
+                     parameters=[],
+                     responseMessages=[],
+                     nickname=None,
+                     notes=None,
+                     bp=None):
+        """
+        Registers an API endpoint.
```
Comment (hobbeswalsh) [inline flask_sillywalk/sillywalk.py]: how is this different from "def register(...)" below? why keep both of them?

File: flask_sillywalk/tests/test_sillywalk.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        self.assertEqual(data['models'], {})
+        # test cheese
+        ret = app.test_client().get("/api/v1/cheese.json")
+        data = json.loads(s(ret.data))
+        self.assertEqual(data["resourcePath"], "cheese")
+        self.assertEqual([x.endpoint for x in app.url_map.iter_rules()],
+                         ['resources',
+                          'cheese',
+                          'blueish.get_cheese',
+                          'static'])
```
Comment (hobbeswalsh) [inline flask_sillywalk/tests/test_sillywalk.py]: thanks for the tests.

File: flask_sillywalk/sillywalk.py (hunk 1 above)
Comment (delijati) [inline flask_sillywalk/sillywalk.py]: To support dynamic adding of urls. Its the same like ``app.add_url_rule('/', 'index', index)`` and ``@app.route('/')``

File: flask_sillywalk/tests/test_sillywalk.py (hunk 2 above)
Comment (delijati) [inline flask_sillywalk/tests/test_sillywalk.py]: There is now nearly 100% coverage so it should be easier to add new features without breaking something.

**Pull request merged**